src/parser/permit-generation-module.ts

```diff
@@ -113,6 +117,81 @@ export class PermitGenerationModule implements Module {
     return result;
   }
 
+  /**
+   * Applies fees to the final result.
+   * How it works:
+   * 1. Fee (read from ENV variable) is subtracted from all of the final result items (user.total, user.task.reward, user.comments[].reward)
+   * 2. Total fee is calculated
+   * 3. A new item is added to the final result object, example:
+   * ```
+   * {
+   *   ...other items
+   *   "ubiquibot-treasury": {
+   *     total: 10.00,
+   *     userId: 1
+   *   }
+   * }
+   * ```
+   * This method is meant to be called before the final permit generation.
+   * @param result Result object
+   * @param erc20RewardToken ERC20 address of the reward token
+   * @returns Result object
+   */
+  async _applyFees(result: Result, erc20RewardToken: string): Promise<Result> {
+    // validate fee related env variables
+    if (!process.env.PERMIT_FEE_RATE) {
+      console.log("PERMIT_FEE_RATE is not set, skipping permit fee generation");
+      return result;
+    }
+    if (!process.env.PERMIT_TREASURY_GITHUB_USERNAME) {
+      console.log("PERMIT_TREASURY_GITHUB_USERNAME is not set, skipping permit fee generation");
```

Comment:
What is this?

Reply:
Treasury must be a separate github account registered with `/wallet`.